### ISSUE TITLE
Go to declaration issue

### ISSUE THREAD

**Issue by zjhmale** [rikvdkleij/intellij-haskell#34]
when i press `⌘-b` on OSX, it will jump to other usage place of a symbol instead of its declaration on the top level, i use IDEA 15.0.2 Build #IC-143.1184.

**Comment by rikvdkleij** [comment]
That indicates it can not find top level declaration, which could be caused it does not get `info` from ghc-mod.

Can you show me code?

**Comment by zjhmale** [comment]
maybe it's a ghc-mod issue, i will construct a issue case later :)

**Comment by zjhmale** [comment]
![report](https://cloud.githubusercontent.com/assets/6234553/14759548/c0f5a540-0959-11e6-8569-52c9aab503b5.gif)

@rikvdkleij as shown in the GIF, when i type `⌘-b` on some symbols, it will jump to a usage place nearby or the definition on the top level, sometimes it just don't go anywhere. really wired. and there is no error like `Error in communication with ghc-mod: Broken pipe.` occurs.

**Comment by zjhmale** [comment]
maybe `ghc-modi` is needed?

**Comment by zjhmale** [comment]
a little complicated here https://github.com/rikvdkleij/intellij-haskell/blob/master/src/main/scala/intellij/haskell/navigate/HaskellReference.scala 😂

**Comment by rikvdkleij** [comment]
Complexity is because `ghc-mod` does not give `info` about locally scoped identifiers and not exported identifiers. I that case I try to find declaration by means of IntelliJ PSI tree. That solution is not completely robust. 

You probably did not export `a`, `b`,.. 
Thanks for your example code. I will improve that code.

**Comment by zjhmale** [comment]
my example code works well with
https://github.com/Atsky/haskell-idea-plugin and https://github.com/carymrobbins/intellij-haskforce but the former one did not use grammar-kit and maybe it is deprecated for now,  we can reference the PSI structure with [haskforce](https://github.com/carymrobbins/intellij-haskforce/blob/master/src/com/haskforce/Haskell.bnf)  :)

**Comment by rikvdkleij** [comment]
Did you export `a` and `b`?

**Comment by zjhmale** [comment]
nope, but the usage and definition are in the same file, it is a little wired to export symbols just for reference even they are in the same file with their definitions.

**Comment by zjhmale** [comment]
still get the issue after export `a` and `b`, the full source is here

```haskell
module Lib (someFunc, b, a) where

import qualified Data.Map as M

someFunc :: IO ()
someFunc = putStrLn "someFunc"

data Foo = Foo [Int] Int
  deriving (Show)

a :: Int
a = 3

b :: Int
b = 3

c :: Foo
c = Foo [a] a

d :: Int
d = a + b

bar :: M.Map String Foo
bar = M.fromList
  [("foo", Foo [a, a] b),
   ("bar", Foo [b] a)]
```

**Comment by rikvdkleij** [comment]
Fixed in 1.0.0.beta

**Comment by zjhmale** [comment]
@rikvdkleij good work! i'll try the new release as soon as possible.
